docs/query-builder.md

```diff
@@ -448,64 +448,136 @@ DB::collection('items')
+  ->where('name', 'John')
+  // Push multiple values to the items array
+  ->push('items', ['hat', 'jeans']);
+// Result:
+// items: ['boots', 'hat', 'jeans']
 
+// Or
+
+// Push the values directly to a model object
 $user->push('items', 'boots');
+$user->push('items', ['hat', 'jeans']);
 ```
 
+To add embedded document or array values to the `messages` array, those values must be specified within a list array.
+
 ```php
 DB::collection('users')
-    ->where('name', 'John')
-    ->push('messages', [
-        'from' => 'Jane Doe',
-        'message' => 'Hi John',
-    ]);
+  ->where('name', 'John')
+    // Push an embedded document as a value to the messages array
+  ->push('messages', [
+      [ 'from'    => 'Jane Doe', 'message' => 'Hi John' ]
+  ]);
+// Result:
+// messages: [
+//      { from: "Jane Doe", message: "Hi John" }
+//  ]
+
+// Or
 
 $user->push('messages', [
-    'from' => 'Jane Doe',
-    'message' => 'Hi John',
+    [ 'from'    => 'Jane Doe', 'message' => 'Hi John' ]
 ]);
 ```
 
-If you **DON'T** want duplicate items, set the third parameter to `true`:
+If you **DON'T** want duplicate values, set the third parameter to `true`:
 
 ```php
 DB::collection('users')
-    ->where('name', 'John')
-    ->push('items', 'boots', true);
+  ->where('name', 'John')
+  ->push('items', 'boots');
+// Result:
+// items: ['boots']
+
+DB::collection('users')
+  ->where('name', 'John')
+  ->push('items', ['hat', 'boots', 'jeans'], true);
+// Result:
+// items: ['boots', 'hat', 'jeans']
+
+// Or
 
-$user->push('items', 'boots', true);
+$user->push('messages', [
+    [ 'from'    => 'Jane Doe', 'message' => 'Hi John' ]
+]);
+// Result:
+// messages: [
+//      { from: "Jane Doe", message: "Hi John" }
+//  ]
+
+$user->push('messages', [
+    [ 'from'    => 'Jess Doe', 'message' => 'Hi' ],
+    [ 'from'    => 'Jane Doe', 'message' => 'Hi John' ],
+], true);
+// Result:
+// messages: [
+//      { from: "Jane Doe", message: "Hi John" }
+//      { from: "Jess Doe", message: "Hi" }
+//  ]
 ```
 
 **Pull**
 
-Remove an item from an array.
+Remove one or multiple values from the `items` array.
 
 ```php
+// items: ['boots', 'hat', 'jeans']
+
 DB::collection('users')
-    ->where('name', 'John')
-    ->pull('items', 'boots');
+  ->where('name', 'John')
+  ->pull('items', 'boots'); // Pull a single value
+// Result:
+// items: ['hat', 'jeans']
 
-$user->pull('items', 'boots');
+// Or pull multiple values
+
+$user->pull('items', ['boots', 'jeans']);
+// Result:
+// items: ['hat']
 ```
 
+Embedded document and arrays values can also be pulled from the `messages` array.
+
 ```php
+// Latest state:
+// messages: [
+//      { from: "Jane Doe", message: "Hi John" }
+//      { from: "Jess Doe", message: "Hi" }
+//  ]
+
 DB::collection('users')
-    ->where('name', 'John')
-    ->pull('messages', [
-        'from' => 'Jane Doe',
-        'message' => 'Hi John',
-    ]);
+  ->where('name', 'John')
+    // It will search for the exact array and pulls it
```

Comment:
```suggestion
    // Pull an embedded document from the array
```

I don't think we need to say "exact" here, as `$pull` performs a match. Quoting [`$pull`: Behavior](https://www.mongodb.com/docs/manual/reference/operator/update/pull/#behavior): 

> If the specified <value> to remove is a document, [$pull](https://www.mongodb.com/docs/manual/reference/operator/update/pull/#mongodb-update-up.-pull) removes only the elements in the array that have the exact same fields and values. The ordering of the fields can differ.

MongoDB still has logic elsewhere where BSON field order factors into comparisons (as I mentioned in https://github.com/mongodb/laravel-mongodb/pull/2685/files#r1406281116), but it's thankfully not a concern with `$pull`. I believe [`$pullAll`](https://www.mongodb.com/docs/manual/reference/operator/update/pullAll/#behavior) is what actually does an exact match with respect to field order. I pray there isn't an extra boolean argument on the `pull()` method that determines which operator to use, so hopefully this won't come up :laughing: 

----

Example from the shell:

```
> db.foo.find()
[
  {
    _id: ObjectId("656758ae5b133c573d501330"),
    x: 1,
    y: [ { a: 1, b: 2, c: 3 } ]
  }

> db.foo.updateOne({ x: 1 }, { $pullAll: { y: [ { a: 1 }]}})
{
  acknowledged: true,
  insertedId: null,
  matchedCount: 1,
  modifiedCount: 0,
  upsertedCount: 0
}

> db.foo.updateOne({ x: 1 }, { $pull: { y: {a: 1}}})
{
  acknowledged: true,
  insertedId: null,
  matchedCount: 1,
  modifiedCount: 1,
  upsertedCount: 0
}

> db.foo.find()
[ { _id: ObjectId("656758ae5b133c573d501330"), x: 1, y: [] } ]
```